cmd/sovereignnode/incomingHeader/incomingSCRs.go

```diff
@@ -0,0 +1,82 @@
+package incomingHeader
+
+import (
+	"encoding/hex"
+	"math/big"
+
+	"github.com/multiversx/mx-chain-core-go/core"
+	"github.com/multiversx/mx-chain-core-go/data"
+	"github.com/multiversx/mx-chain-core-go/data/smartContractResult"
+	"github.com/multiversx/mx-chain-go/process"
+)
+
+const (
+	minTopicsInEvent  = 4
+	numTransferTopics = 3
+)
+
+type scrInfo struct {
+	scr  *smartContractResult.SmartContractResult
+	hash []byte
+}
+
+func (ihp *incomingHeaderProcessor) createIncomingSCRs(events []data.EventHandler) ([]*scrInfo, error) {
+	scrs := make([]*scrInfo, 0, len(events))
+
+	for _, event := range events {
+		topics := event.GetTopics()
+		if len(topics) < minTopicsInEvent || len(topics[1:])%numTransferTopics != 0 {
```

Comment:
what does this line test/enforce?

Reply:
It enforces the topics to be correctly formatted, as described in the PR description:
> Topics = address, LIST<tokenID, nonceAsBytes, valueAsBytes>

Reply:
it partially enforces the topics to be correctly formatted, it just checks that it has a number, not that the topics are actually the ones needed.